src/Microsoft.DotNet.XHarness.Android/AdbRunner.cs

```diff
@@ -423,28 +453,25 @@ public List<string> PullFiles(string devicePath, string localPath)
             catch (Exception toLog)
             {
                 logger.LogError(toLog, $"Exception thrown while trying to find compatible device with {propertyName} {apkRequiredProperty}");
-                return null;
+                return new Dictionary<string, string>();
             }
 
             if (allDevicesAndTheirProperties.Count == 0)
             {
                 logger.LogError("No attached device detected");
-                return null;
+                return new Dictionary<string, string>();
             }
 
-            if (allDevicesAndTheirProperties.Any(kvp => apkRequiredProperty.Equals(kvp.Value, StringComparison.OrdinalIgnoreCase)))
-            {
-                // Key-value tuples here are of the form <device serial number, device property>
-                KeyValuePair<string, string?> firstAvailableCompatible = allDevicesAndTheirProperties.FirstOrDefault(kvp => apkRequiredProperty.Equals(kvp.Value, StringComparison.OrdinalIgnoreCase));
-                logger.LogDebug($"Using first-found compatible device of {allDevicesAndTheirProperties.Count} total- serial: '{firstAvailableCompatible.Key}' - {propertyName}: {firstAvailableCompatible.Value}");
-                return firstAvailableCompatible.Key;
-            }
-            else
+            var result = allDevicesAndTheirProperties
+                .Where(kvp => !string.IsNullOrEmpty(kvp.Value) && kvp.Value.Split().Contains(apkRequiredProperty))
+                .ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
+            
+            if (result.Count() == 0)
             {
                 // In this case, the enumeration worked, we found one or more devices, but nothing matched the APK's architecture; fail out.
                 logger.LogError($"No devices found with {propertyName} '{apkRequiredProperty}'.");

```

Comment:
I would be more explicit here and say that we found devices but none with the property. I know it means that but I would be overexplicit in these cases